src/mpi/coll/src/csel.c

```diff
@@ -645,6 +656,13 @@ static csel_node_s *prune_tree(csel_node_s * root, MPIR_Comm * comm_ptr)
                     node = node->failure;
                 break;
 
+            case CSEL_NODE_TYPE__OPERATOR__COMM_AVG_PPN_LE:
+                if (node->u.comm_avg_ppn_le.val * comm_ptr->node_count <= comm_ptr->local_size)
+                    node = node->success;
```

Comment:
If I have like, for a communicator which has 8 ranks on 2 nodes, and with json file: 
"ppn<=1"
"ppn<=2"
"ppn<=4"
Will it always hit the first branch?

Reply:
Oh, I got the equation wrong.  It should be:

`comm_ptr->local_size <= node->u.comm_avg_ppn_le.val * comm_ptr->node_count`

Fixed.